src/test/java/de/bonndan/nivio/security/CustomOAuth2UserServiceTest.java

```diff
@@ -16,51 +20,84 @@
 
     private OAuth2User oAuth2User;
 
-    private String userName = "Mary";
+    private String name = "Mary";
     private String login = "foo";
     private String avatarUrl = "https://www.avatar.com";
-
+    private String externalId = "123";
+    private String idp = "github";
     private Collection<OAuth2UserAuthority> authorities;
+    private CustomOAuth2User customOAuth2User;
 
+    @MockBean
+    AppUserRepository appUserRepository;
 
     @BeforeEach
     public void setup() {
         oAuth2User = mock(OAuth2User.class);
         when(oAuth2User.getAttribute("login")).thenReturn(login);
         when(oAuth2User.getAttribute("avatar_url")).thenReturn(avatarUrl);
-        when(oAuth2User.getAttribute("id")).thenReturn(123);
+        when(oAuth2User.getAttribute("id")).thenReturn(externalId);
+        when(oAuth2User.getAttribute("name")).thenReturn(name);
+
         when(oAuth2User.getAttributes()).thenReturn(Map.of());
 
         Map<String, Object> authorityAttributes = Map.of("key", new Object());
         OAuth2UserAuthority grantedAuthority = new OAuth2UserAuthority(authorityAttributes);
         authorities = List.of(grantedAuthority);
         doReturn(authorities).when(oAuth2User).getAuthorities();
+
     }
 
     @Test
     void fromGitHubUser() {
 
         //given
-        when(oAuth2User.getAttribute("name")).thenReturn(userName);
-
-        //when
-        CustomOAuth2User customOAuth2User = CustomOAuth2UserService.fromGitHubUser(oAuth2User, "login", "name");
+        when(oAuth2User.getAttribute("name")).thenReturn(name);
+        customOAuth2User = CustomOAuth2UserService.fromGitHubUser(oAuth2User, "login", "name");
 
         //then
         assertThat(customOAuth2User).isNotNull();
         assertThat(customOAuth2User.getAlias()).isEqualTo(login);
+        assertThat(customOAuth2User.getExternalId()).isEqualTo(externalId);
+        assertThat(customOAuth2User.getName()).isEqualTo((name));
     }
 
     @Test
     void fromGitHubUserWithMissingNameFallsBackToLogin() {
 
         //given
         when(oAuth2User.getAttribute("name")).thenReturn(null);
-
-        //when
-        CustomOAuth2User customOAuth2User = CustomOAuth2UserService.fromGitHubUser(oAuth2User, "login", "name");
+        customOAuth2User = CustomOAuth2UserService.fromGitHubUser(oAuth2User, "login", "name");
 
         //then
         assertThat(customOAuth2User.getName()).isEqualTo(login);
     }
+
+    @Test
```

Comment:
//when is missing and the repo is not involved in this test

* make sure the repo is used using `verify `
* use `ArgumentCaptor` to check the saved value